web/app/scripts/toddow/toddow-directive.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * This directive creates a ToDDoW calendar for aggregating information event occurrence.
+ *  The bound variable `chartData` needs only be a list of objects which contain the property
+ *  'occurred_from'.
+ */
+
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function ToDDoW($window) {
+        var module = {
+            restrict: 'A',
+            scope: {
+              chartData: '='
+            },
+            template: '<svg></svg>',
+            link: function(scope, elem, attrs) {
+                var d3 = $window.d3;
+                var moment = $window.moment;
+                var rawSvg = elem.find('svg')[0];
+                var cellSize = 30,
+                    height = 475,
+                    width = 752;
+                var rect, color, svg, tooltip;  // GLOBAL
+                init();
+
+                /**
+                 * Watch for changes to chartData and redraw and redraw and redraw
+                 */
+                scope.$watch('chartData', function(val) {
+                    if (val) {
+                        color = d3.scale.linear()
```

Comment:
Ah, there appears to be some code I meant to change - this piece should be in the $watch block so that we can update the domain to match the dataset we're dealing with (0 to max rather than 0 to 4)